lib/adapter.js

```diff
@@ -146,3 +154,17 @@ adapterFactory.run = async function (cid, config, specs, capabilities) {
 
 export default adapterFactory
 export { CucumberAdapter, adapterFactory }
+
+function isAsync (func) {
+    const string = func.toString().trim()
+    return !!(
+        // native
+        string.match(/^async /) ||
+        // babel (this may change, but hey...)
+        string.match(/return _ref[^.]*\.apply/) ||
```

Comment:
it didn't work for me originally.. but I tried it now with just the fn.name === 'async' check and it passes the tests. Unless I'm missing some edge case I forgot about - you might be right and that check might be enough, removed this code now